src/deepqmc/gnn/electron_gnn.py

```diff
@@ -332,10 +332,10 @@ def __init__(
         ghost_coords=None,
     ):
         super().__init__()
-        n_nuc, n_up, n_down = mol.n_particles
+        n_nuc, n_up, n_down = hamil.mol.n_nuc, hamil.n_up, hamil.n_down
```

Comment:
Now it's a little odd that `n_up` and `n_down` is on the Hamiltonian, but `n_nuc` is on the Molecule. Maybe move everything to the Hamiltonian?

Reply:
I agree, I've moved `n_nuc` to Hamiltonian in the last commit